### PR TITLE
🔧 Correct DeprecationWarning config and comment in pytest settings

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -140,8 +140,8 @@ junit_family = "xunit2"
 filterwarnings = [
     "error",
     'ignore:"@coroutine" decorator is deprecated since Python 3\.8, use "async def" instead:DeprecationWarning',
-    # TODO: needed by AnyIO in Python 3.9, try to remove after an AnyIO upgrade
-    'ignore:The loop argument is deprecated since Python 3\.8, and scheduled for removal in Python 3\.10:DeprecationWarning',
+    # TODO: needed by asyncio in Python 3.9.7 https://bugs.python.org/issue44815, try to remove on 3.9.8
+    'ignore:The loop argument is deprecated since Python 3\.8, and scheduled for removal in Python 3\.10:DeprecationWarning:asyncio',
```
Comment (graingert) [inline pyproject.toml:144]: this `:asyncio` bit means it only hides the warning if the warning is raised by something in `asyncio` and not from elsewhere, eg `fastapi`

Comment (graingert) [reply pyproject.toml:144]: it's actually not pytest's fault: https://docs.python.org/3/library/warnings.html#describing-warning-filters

File: pyproject.toml [hunk 2]
```diff
@@ -140,8 +140,8 @@ junit_family = "xunit2"
 filterwarnings = [
     "error",
     'ignore:"@coroutine" decorator is deprecated since Python 3\.8, use "async def" instead:DeprecationWarning',
-    # TODO: needed by AnyIO in Python 3.9, try to remove after an AnyIO upgrade
-    'ignore:The loop argument is deprecated since Python 3\.8, and scheduled for removal in Python 3\.10:DeprecationWarning',
+    # TODO: needed by asyncio in Python 3.9.7 https://bugs.python.org/issue44815, try to remove on 3.9.8
```
Comment (graingert) [inline pyproject.toml:143]:
```suggestion
    # TODO: needed by asyncio in Python 3.9.7 https://bugs.python.org/issue45097, try to remove on 3.9.8
```

File: pyproject.toml [hunk 3]
```diff
@@ -140,8 +140,8 @@ junit_family = "xunit2"
 filterwarnings = [
     "error",
     'ignore:"@coroutine" decorator is deprecated since Python 3\.8, use "async def" instead:DeprecationWarning',
```
Comment (graingert) [inline pyproject.toml:142]:
```suggestion
```

**Comment by tiangolo** [comment]
Nice, thank you @graingert! :rocket: :heavy_check_mark:

**Pull request merged**